tests/Google/Service/ResourceTest.php

```diff
@@ -47,16 +47,36 @@ public function __construct(Client $client)
   }
 }
 
-class TestMediaTypeStream extends Stream
-{
-  public $toStringCalled = false;
 
-  public function __toString()
-  {
-    $this->toStringCalled = true;
+// if php version is lower than 7.2, then psr7 v1 is installed,
+// so __toString() return is not explicitly typed in parent class
+if(version_compare(PHP_VERSION,'7.2.0','<')) {
+    class TestMediaTypeStream extends Stream
+    {
+        public $toStringCalled = false;
 
-    return parent::__toString();
-  }
+        public function __toString()
+        {
+            $this->toStringCalled = true;
+
+            return parent::__toString();
+        }
+    }
+// version is gte 7.2, so psr7 version can be:
+// * v2, so __toString() return is typed
+// * v1, so we can safely override __toString() to make it return string
+} else {
+    class TestMediaTypeStream extends Stream
+    {
+        public $toStringCalled = false;
+
+        public function __toString():string
```

Comment:
@n00dl3, it looks like this may be the culprit for the 5.6 test failure. Could we re-evaluate this approach?

Reply:
Honestly I have no idea, without putting the class in an external file and require it conditionally like :
```php
// if php version is lower than 7.2, then psr7 v1 is installed,
// so __toString() return is not explicitly typed in parent class
if(version_compare(PHP_VERSION,'7.2.0','<')) {
    require_once __DIR__.'/TestMediaTypeStreamv5+.php';
// version is gte 7.2, so psr7 version can be:
// * v2, so __toString() return is typed
// * v1, so we can safely override __toString() to make it return string
} else {
    require_once __DIR__.'/TestMediaTypeStreamv7+.php';
}
```

Anyway, tests will fail right after that because of the `setUp():void` declaration.

Maybe I should just stick to the ` public function __toString():string` declaration compatible with php 7+ ?

@dwsupplee An opinion, please ?

Reply:
Fixed in 43dd3443602e475951966da3be6745b73f49ec33